src/de.rs

```diff
@@ -594,6 +606,46 @@ impl<'de> de::IntoDeserializer<'de, Error> for Value<'de> {
     }
 }
 
+struct SpannedDeserializer<'a> {
+    start: Option<usize>,
+    end: Option<usize>,
+    value: Option<Value<'a>>,
+}
+
+impl<'de> de::MapAccess<'de> for SpannedDeserializer<'de> {
+    type Error = Error;
+
+    fn next_key_seed<K>(&mut self, seed: K) -> Result<Option<K::Value>, Error>
+    where
+        K: de::DeserializeSeed<'de>,
+    {
+        if self.start.is_some() {
+            seed.deserialize(spanned::START.into_deserializer()).map(Some)
```

Comment:
This needs to be:

```rust
use serde::de::value::BorrowedStrDeserializer;

seed.deserialize(BorrowedStrDeserializer::new(spanned::START)).map(Some)
```

in order for my other comment to work.

Reply:
Fixed, thanks!